src/fs/cluster_info.hh

```diff
@@ -0,0 +1,70 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Copyright (C) 2020 ScyllaDB
+ */
+
+#pragma once
+
+#include "fs/cluster.hh"
+#include "fs/inode.hh"
+#include "fs/units.hh"
+#include <cassert>
+#include <map>
+
+namespace seastar::fs {
+
+// Each cluster_data_vec corresponds to an inode_data_vec
+struct cluster_data_vec {
+    inode_t data_owner;
+    file_range data_range;
+};
+
+struct cluster_info {
```

Comment:
Maybe we could change cluster_info to be a class and make `valid_data_size`, `read_only`, and `data` private? We don't want to modify those fields manually because that could leave the struct in an invalid state. Also, we would like to have some read-only getters for those private fields.